tests/test_forwarder.py

```diff
@@ -4,16 +4,24 @@
 import random
 import select
 import socket
+import sys
 import threading
 import time
-import unittest
+import warnings
 
+import pytest
```

Comment:
Excess external dependence. In this case it is better to use the `@unittest.skipIf` of the standard `unittest` module. And provide `unittest2` for python 2.6 compatibility.